src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -1536,123 +1536,158 @@ private String[] SplitInternal(String separator, String[] separators, Int32 coun
+        [Pure]
+        public unsafe string Trim(params char[] trimChars)
+        {
+            if (null == trimChars || trimChars.Length == 0)
+            {
                 return TrimHelper(TrimBoth);
             }
-            return TrimHelper(trimChars,TrimBoth);
+            fixed (char* pTrimChars = trimChars)
+            {
+                return TrimHelper(pTrimChars, trimChars.Length, TrimBoth);
+            }
         }
-    
+
+        // Removes a set of characters from the beginning of this string.
+        public string TrimStart() => TrimHelper(TrimHead);
+
         // Removes a set of characters from the beginning of this string.
-        public String TrimStart(params char[] trimChars) {
-            if (null==trimChars || trimChars.Length == 0) {
+        public unsafe string TrimStart(char trimChar) => TrimHelper(&trimChar, 1, TrimHead);
+
+        // Removes a set of characters from the beginning of this string.
+        public unsafe string TrimStart(params char[] trimChars)
+        {
+            if (null == trimChars || trimChars.Length == 0)
+            {
                 return TrimHelper(TrimHead);
             }
-            return TrimHelper(trimChars,TrimHead);
+            fixed (char* pTrimChars = trimChars)
+            {
+                return TrimHelper(pTrimChars, trimChars.Length, TrimHead);
+            }
         }
 
-        // Removes a set of characters from the beginning of this string.
-        public String TrimStart() => TrimHelper(TrimHead);
-        
+        // Removes a set of characters from the end of this string.
+        public string TrimEnd() => TrimHelper(TrimTail);
+
+        // Removes a set of characters from the end of this string.
+        public unsafe string TrimEnd(char trimChar) => TrimHelper(&trimChar, 1, TrimTail);
 
         // Removes a set of characters from the end of this string.
-        public String TrimEnd(params char[] trimChars) {
-            if (null==trimChars || trimChars.Length == 0) {
+        public unsafe string TrimEnd(params char[] trimChars)
+        {
+            if (null == trimChars || trimChars.Length == 0)
+            {
                 return TrimHelper(TrimTail);
             }
-            return TrimHelper(trimChars,TrimTail);
+            fixed (char* pTrimChars = trimChars)
+            {
+                return TrimHelper(pTrimChars, trimChars.Length, TrimTail);
+            }
         }
-        
-        // Removes a set of characters from the end of this string.
-        public String TrimEnd()=> TrimHelper(TrimTail);
-        
-
-        // Trims the whitespace from both ends of the string.  Whitespace is defined by
-        // Char.IsWhiteSpace.
-        //
-        [Pure]
-        public String Trim() {
-            Contract.Ensures(Contract.Result<String>() != null);
-            Contract.EndContractBlock();
 
-            return TrimHelper(TrimBoth);        
-        }
+        private string TrimHelper(int trimType)
```

Comment:
I'd personally change the 3 const ints TrimHead, TrimTrail, TrimBoth, into enum ... given you're already cleaning up and refactoring the code ... just a thought ...